packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -525,6 +526,11 @@ export interface IUserPool extends IResource {
    */
   readonly userPoolArn: string;
 
+  /**
+   * Get all identity providers registered with this user pool.
+   */
+  readonly identityProviders: IUserPoolIdentityProvider[];
```

Comment:
Wonder if this couldn't/shouldn't have just been a list of `UserPoolClientIdentityProvider`s.

I guess not because of the `Client` in there... would have been simpler to have been a list of strings... but no worries I guess.

Reply:
BTW I think I feel more for setting this list upon `UserPool.fromAttributes()` than doing `UserPool.fromName()` and calling `registerIdentityProvider()` multiple times.

Since it's not really *mutating* but rather giving the construct lib more information about the thing that's already there, feels like it should be a part of the import.